torch/distributed/elastic/multiprocessing/api.py

```diff
@@ -234,24 +234,19 @@ class DefaultLogsSpecs(LogsSpecs):
     """
     def __init__(
         self,
-        log_dir: Optional[str] = os.devnull,
+        log_dir: Optional[str] = None,
         redirects: Union[Std, Dict[int, Std]] = Std.NONE,
         tee: Union[Std, Dict[int, Std]] = Std.NONE,
         local_ranks_filter: Optional[Set[int]] = None,
     ) -> None:
-
-        log_dir = log_dir or os.devnull
-
         if log_dir != os.devnull:
-            if not os.path.exists(log_dir):
+            if not log_dir:
+                log_dir = tempfile.mkdtemp(prefix="torchelastic_")
```

Comment:
I expect this is fine in most cases -- though with this temp directory we'll never clean up the logs which could cause issues. Should we be deleting this when elastic shuts down?

Reply:
it was behaving this way before refactoring changes. Btw, users can now define their own logic using custom logspecs (using entrypoints)